### PR TITLE
Feature WebInterface

### LINKED ISSUE #87: Web Interface - Server Communication
The web interface clients need to communicate with the server to properly interact with the game.

This is a subtask of #43 - Web Interface

### PR DESCRIPTION
**Pull request by Taeir**
# Web Interface
#### Changes

It is finally here, the basis of our Web Interface!
- Added an embedded webserver, jetty
- Added classes to manage and use that webserver
- Added HTML and JavaScript code for the webserver
- Added Servlets (in java) to handle the server side of the web code
- Added methods to export a Level as JSON Object.
- WebServer will be started when the game is started, on port 8080.
- More web related stuff
#### Static Analysis

**Coverage**
97% line coverage
Test coverage report

**Checkstyle:**
All checkstyle warnings are about Import order. These errors will go away when the code is merged into master.
Checkstyle report

**PMD:**
There are PMD warnings about the test classes, because the same string occurs in them a lot. I suggest disabling this PMD warning for test classes, as we do not want to put everything into constants.
PMD report

**FindBugs:**
No warnings
#### For review

Relevant classes are those in the webinterface package (both src/main and src/test), the Main class and the Level class.
#### Tasks

This involves tasks #43, #85, #86, #87, #88.

This closes #87.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/webinterface/WebClient.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package nl.tudelft.contextproject.webinterface;
+
+/**
+ * Class to represent connected web clients.
+ */
+public class WebClient {
+	private Boolean team;
```
Comment (BCrlrd) [inline src/main/java/nl/tudelft/contextproject/webinterface/WebClient.java]: Add comment explaining why it's logical for this to be a boolean

File: src/main/resources/webinterface/main.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            showError("Something went wrong: [" + status + "] " + data);
+        }
+    });
+}
+
+/**
+ * Checks if the given data implies that we are authorized.
+ * If not, this method will attempt to authenticate and will return false.
+ */
+function checkAuthorized(data) {
```
Comment (Geweldig) [inline src/main/resources/webinterface/main.js:54]: All other methods have a param tag, why doesn' this one?

File: src/main/resources/webinterface/main.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+        //Check if we are authorized
+        if (!checkAuthorized(data)) return;
+        
+        //Update the map
+        updateExplored(data);
+    }, "json");
+}
+
+function updateMap(data) {
```
Comment (Geweldig) [inline src/main/resources/webinterface/main.js:266]: no comment?

File: src/main/resources/webinterface/main.js [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+                //Update the className
+                cell.className = getClassForTileType(data.tiles[x][y]);
+            }
+        }
+        
+        gMap = data;
+    }
+}
+
+function updateExplored(data) {
```
Comment (Geweldig) [inline src/main/resources/webinterface/main.js:322]: no comment?

File: src/main/resources/webinterface/main.js [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+        
+        for (i = 0; i < row.length; i++) {
+            $(document.getElementById("y" + row[i] + "x" + x)).addClass("explored");
+        }
+    }
+    
+    gExplored = data;
+}
+
+function getClassForTileType(tileType) {
```
Comment (Geweldig) [inline src/main/resources/webinterface/main.js:341]: no comment?

**Comment by Geweldig** [comment]
Looks good to me now!

**Comment by ChielBruin** [comment]
Same here!

**Pull request merged**